src/main/java/io/fusionauth/http/util/HTTPTools.java

```diff
@@ -119,8 +119,10 @@ public static boolean isURICharacter(byte ch) {
     return ch >= '!' && ch <= '~';
   }
 
+  // RFC9110 section-5.5 allows for "obs-text", which includes 0x80-0xFF, but really shouldn't be used.
   public static boolean isValueCharacter(byte ch) {
-    return isURICharacter(ch) || ch == ' ' || ch == '\t' || ch == '\n';
+    int intVal = ch & 0xFF;  // Convert the value into an integer without extending the sign bit.
+    return isURICharacter(ch) || intVal == ' ' || intVal == '\t' || intVal == '\n' || intVal >= 0x80;
```

Comment:
In addition to header values, this method is used when parsing the response status message. Does the RFC allow for this type of character there as well? Or is there any risk to using this same logic for both of these use cases?

Reply:
That part hasn't been superseded by RFC9110, so according to RFC7230, the status reason can still (unfortunately) contain `obs-text`.

> The reason-phrase element exists for the sole purpose of providing a
>    textual description associated with the numeric status code, mostly
>    out of deference to earlier Internet application protocols that were
>    more frequently used with interactive text clients.  A client SHOULD
>    ignore the reason-phrase content.
> 
>      reason-phrase  = *( HTAB / SP / VCHAR / obs-text )